### ISSUE TITLE
Using PGP/MIME for signing

### ISSUE THREAD

**Issue by timorl** [mailpile/Mailpile#1932]
I want to use PGP/MIME for signing my email, so it doesn't look scary to people who don't know what this means. I thought that the option "Prefer compatibility, avoid PGP/MIME" would be what controls this behaviour, but it doesn't seem to be working for me. Whether it is checked or unchecked all the emails are sent signed the "traditional" way. I have also tried checking and unchecking the option "Minimize metadata" (as the only other one that could conceivably influence how email looks), but it didn't do anything noticable.

Am I doing something wrong? Has PGP/MIME support been removed while I wasn't looking?

I am running Mailpile v1.0.0rc1, build on gentoo from the mrueg overlay about two weeks ago (afaik this is simply the version compiled from github). I'm pretty sure it worked before that update.

**Comment by BjarniRunar** [comment]
This is deliberate, working as intended! Signature crap at the bottom of a text part is way less scary than random attachments that people are unable to open.

See #710 for part of the rationale, as well as #1693. :-)

I'm going to leave this issue open, as I am aware that there are people (such as @JocelynDelalande) who disagree with my decision. I encourage people to try and convince me I'm wrong, but I do request you read the old issues and consider the arguments made there first.

**Comment by timorl** [comment]
I see, thank you for the quick response.

I read the issues you linked. I'm not sure I agree that attachments are more confusing than random text -- I'd expect the opposite, but among the people I mailed no one complained about either, so I don't even have anecdotal data. Inline signing certainly looks ugly in quoted emails and in certain situations mailpile warns me that someone quoting my message sent an invalid signature (when they don't use '>' for quotations I think?), but I can live with that and will probably leave inline signing on.

I also noticed PR  #1930 and I feel that the solution there is a nice compromise. I got quite confused by the way the options were and adding an explicit "Prefer PGP/MIME" checkbox would certainly fix that. Since it wouldleave the default setting at it is now and be in the "too many encryption settings" section I don't think it would impact the way most people use mailpile too much.

**Comment by BjarniRunar** [comment]
@timorl The main difference is, attachments are "important documents", junk in the signature is just junk in the signature. Being unable to process the former is much more worrying for a recipient than the latter.

Regarding PR #1930, I balk at this for a few reasons. One is that every new setting makes things more confusing, both for the user and for the people supporting the software (as there are more combinations of settings that need to be checked during troubleshooting). This applies to all settings, so every time someone says "just add a setting", my knee jerk reaction is negative.

The other is more philosophical: as I am convinced that forcing PGP/MIME all the time has multiple downsides and no upsides, why would I add a setting that just makes things worse?  This is where people have the opportunity to convince me otherwise, or simply convince me that there is popular demand and doing what people want is more important than doing what is "right."

**Comment by timorl** [comment]
Count me as a person who would like that option then.

Although I think my main problem is that now the "Prefer compatibility, avoid PGP/MIME" option is confusing by itself. Adding an "Always use PGP/MIME" option (or better yet, a choice between "Never", "Use only with non-plaintext content(default)" and "Always") would clear things up in the UI. On the other hand I sympathize with your argument against adding options. With that in mind, removing the "Prefer compatibility..." option would also _kind of_ solve part of the problem, but I suspect this is a very bad idea. Maybe renaming it would help, but I have no idea what would be a better name.

**Comment by BjarniRunar** [comment]
@timorl I do sympathize. My problem is I have two sets of users to contend with. Ones like you, who understand the tech and are looking for certain technical terms and have certain expectations - and users who do not know the lingo and just want things to work.

The settings currently target the latter audience whenever possible. Also, it's a feature not a bug that the implementation of "prefer compatibility" may change over time. As the app improves and we get batter at compatibility, that setting WILL do new things.

That's only a problem for techies who understand "too much."

**Comment by BjarniRunar** [comment]
Note that I don't want Mailpile to become another enigmail. The UI will never expose all the features of PGP using all the PGP lingo. People who want that should either use enigmail or learn to use the Mailpile CLI, where you can really tweak things to your heart's content.

**Comment by Kabouik** [comment]
I'm not a techie, although I'm curious about techie stuff and am keen on documenting myself so I might not be the best example of the second audience either, but I must admit that sometimes the wording is confusing to me as well because I am not sure what is exactly meant with some options. Since it's not the exact lingo and since there is no abundant documentation on Mailpile, I can't really look for what is behind the options, be it in Mailpile-specific resources or just anywhere using keywords and web searches. I remember asking you several questions about these encryption options when I first installed Mailpile.

I don't know if I'm alone in that case, but my personal belief is it might be less ambiguous and easier to understand for both audience if the exact wording was more often used and supplemented with mouse-over tooltips to explain things in Layman's terms.

**Comment by BjarniRunar** [comment]
There is now a "Prefer PGP/MIME" option in accounts... but only if you've enabled developer settings on the preferences page. This is my compromise, I hope people aren't too unhappy!

**Comment by BjarniRunar** [comment]
@ryantate Feel free to enable the developer options and change your settings if you feel strongly.

Your "why" questions have already been answered elsewhere. And yes, I am honestly convinced that the ugly message is less confusing than the elegant one with they mysterious attachment. Normal people perceive attachments as important. Garbage signatures, not so much.

I understand you're worried about how your e-mails look - but pretty and functional are sometimes at odds. If the STANDARDS and the way other e-mail clients are doing PGP is so great, then why is (virtually) nobody encrypting their mail? The community is wrong about a lot of stuff, and if I weren't arrogant enough to think I can do better, I wouldn't be working on this project in the first place. :grin:

**Comment by ryantate** [comment]
Thanks for the reply. I do enjoy Mailpile and thank you for creating it. Despite frustration with this one issue.

I do have developer options enabled and have checked "Prefer PGP/MIME". But it seems to not be using PGP/MIME. As you seem to state right here in this thread, this is by design: "Inline signatures are almost exclusively used for messages with only a single text/plain part."

Is there any way whatsoever for me to get Mailpile to use PGP/MIME "for messages with only a single text/plain part?" You tell me "to enable the developer options and change your settings" so I assume there must be.

**Comment by BjarniRunar** [comment]
Those setting should suffice. Note that you need to enable that for every account. If it still doesn't work, there might be a bug somewhere!
